tests/framework/di/InstanceTest.php

```diff
@@ -41,9 +43,81 @@ public function testEnsure()
 
         $this->assertTrue(Instance::ensure('db', 'yii\db\Connection', $container) instanceof Connection);
         $this->assertTrue(Instance::ensure(new Connection, 'yii\db\Connection', $container) instanceof Connection);
-        $this->assertTrue(Instance::ensure([
+        $this->assertTrue(Instance::ensure(['class' => 'yii\db\Connection', 'dsn' => 'test'], 'yii\db\Connection', $container) instanceof Connection);
+    }
+
+    public function testEnsureWithoutType()
+    {
+        $container = new Container;
+        $container->set('db', [
+            'class' => 'yii\db\Connection',
+            'dsn' => 'test',
+        ]);
+
+        $this->assertTrue(Instance::ensure('db', null, $container) instanceof Connection);
+        $this->assertTrue(Instance::ensure(new Connection, null, $container) instanceof Connection);
+        $this->assertTrue(Instance::ensure(['class' => 'yii\db\Connection', 'dsn' => 'test'], null, $container) instanceof Connection);
+    }
+
+    public function testEnsureMinimalSettings()
+    {
+        Yii::$container->set('db', [
             'class' => 'yii\db\Connection',
             'dsn' => 'test',
-        ], 'yii\db\Connection', $container) instanceof Connection);
+        ]);
+
+        $this->assertTrue(Instance::ensure('db') instanceof Connection);
+        $this->assertTrue(Instance::ensure(new Connection) instanceof Connection);
+        $this->assertTrue(Instance::ensure(['class' => 'yii\db\Connection', 'dsn' => 'test']) instanceof Connection);
+
+        Yii::$container = new Container;
+    }
+
+    public function testCheckException()
+    {
+        $container = new Container;
+        $container->set('db', [
+            'class' => 'yii\db\Connection',
+            'dsn' => 'test',
+        ]);
+
+        try {
```

Comment:
It should be done with separated test cases and `$this->setExpectedException`, because current test will pass successfully when no exception was thrown.